Spec/RealtimeClientChannel.swift

```diff
@@ -1725,167 +1725,228 @@ class RealtimeClientChannel: QuickSpec {
+                                expect((client.transport as! TestProxyTransport).protocolMessagesSent.filter({ $0.action == .message })).to(haveCount(1))
                             }
-                            expect((client.transport as! TestProxyTransport).protocolMessagesSent.filter({ $0.action == .message })).to(haveCount(1))
-                        }
-                    }
-
-                    // RTL6c2
-                    context("the message should be queued and delivered as soon as the connection state returns to CONNECTED if the connection is") {
-                        let options = AblyTests.commonAppSetup()
-                        options.useTokenAuth = true
-                        options.autoConnect = false
-                        var client: ARTRealtime!
-                        var channel: ARTRealtimeChannel!
-
-                        beforeEach {
-                            client = ARTRealtime(options: options)
-                            channel = client.channels.get("test")
-                            expect(client.options.queueMessages).to(beTrue())
                         }
-                        afterEach { client.close() }
 
-                        func publish(_ done: @escaping () -> ()) {
-                            channel.publish(nil, data: "message") { error in
-                                expect(error).to(beNil())
-                                expect(client.connection.state).to(equal(ARTRealtimeConnectionState.connected))
-                                done()
+                        it("INITIALIZED then the messages should be published immediately") {
+                            let client = AblyTests.newRealtime(AblyTests.commonAppSetup())
+                            defer { client.dispose(); client.close() }
+                            let channel = client.channels.get("test")
+                            waitUntil(timeout: testTimeout) { done in
+                                client.connection.once(.connected) { _ in
+                                    done()
+                                }
                             }
-                        }
+                            expect(channel.state).to(equal(ARTRealtimeChannelState.initialized))
+                            expect(client.connection.state).to(equal(ARTRealtimeConnectionState.connected))
 
-                        it("INITIALIZED") {
                             waitUntil(timeout: testTimeout) { done in
-                                expect(client.connection.state).to(equal(ARTRealtimeConnectionState.initialized))
-                                publish(done)
-                                client.connect()
-                                expect(channel.queuedMessages).to(haveCount(1))
+                                channel.publish(nil, data: "message") { error in
+                                    expect(error).to(beNil())
+                                    done()
+                                }
```

Comment:
This test doesn't seem to test that the message is actually published immediately (ie without waiting for the channel to attach) -- it just checks that the publish completes eventually, you're not checking channel state after the publish completed. (similarly with the other tests)

Reply:
Fixed a93a589